addons/azureSets/azuresets.lua

```diff
@@ -68,19 +68,36 @@ windower.register_event('login', initialize)
 
 windower.register_event('job change', initialize:cond(function(job) return job == 16 end))
 
-function set_spells(spellset)
+function set_spells(spellset, setmode)
     if windower.ffxi.get_player()['main_job_id'] ~= 16 --[[and windower.ffxi.get_player()['sub_job_id'] ~= 16]] then return nil end
     if settings.spellsets[spellset] == nil then return end
-    if settings.spellsets[spellset]:equals(get_current_spellset()) then
+    if is_spellset_equipped(settings.spellsets[spellset]) then
         error(spellset..' was already equipped.')
         return
     end
 
     log('Starting to set '..spellset..'.')
-    set_spells_from_spellset(spellset, 'remove')
+    if setmode=='clearfirst' or (setmode == nil and settings.setmode == 'clearfirst') then
+        remove_all_spells(nil)
+        windower.send_command('@wait .65;lua i azuresets set_spells_from_spellset '..spellset..' add')
```

Comment:
I like this. I might rewrite the function itself to do this instead, mostly because I hate the send_command anyway. I'll test it out at least.